### DISCUSSION TITLE
Phase field coupled mechanical stress

### DISCUSSION THREAD

**Discussion by biaogxb** [idaholab/moose#21355] [Q&A Modules: Solid mechanics] [answered]
I want to use the phase field method and coupling the mechanical stress to simulate the growth of dendrites. When the mechanical stress is not added, everything looks normal, but the following errors occur after coupling the mechanical stress
`*** ERROR ***`
`The following error occurred in the object "MOOSE Problem", of type "FEProblem".`

`Material property 'mechanical_strain', requested by 'stress' is not defined on block 0`
`Material property 'mechanical_strain', requested by 'stress_face' is not defined on block 0`
`Material property 'mechanical_strain', requested by 'stress_neighbor' is not defined on block 0`

I will calculate the elastic energy as follows.
![image](https://user-images.githubusercontent.com/95063954/174522526-e4995399-dd95-4991-a658-2313fb6b0ffa.png)

The added code is as follows:
kernel part
`  [./AC_bulk]`
    `type = AllenCahn`
 `   variable = eta`
`    f_name = FA`
`  [../]`
material part
`   # Energy of the barrier`
   `[./Free]`
 `   type = DerivativeParsedMaterial`
  `  f_name = FF`
   ` material_property_names = 'B'`
   ` args='eta'`
   ` function = 'B*eta*eta*(1-eta)*(1-eta)'`
    `derivative_order = 1`
 ` [../]`

`#Mechanics parameeters`
 ` [./Stiffness_matrix]`
  `  type = ComputeElasticityTensor`
   ` C_ijkl = '3294.1 1852.9 1852.9 3294.1 1852.9 3294.1 720.6 720.6 720.6'`
   ` fill_method = symmetric9`
 ` [../]`

`  [./stress]`
   ` type = ComputeLinearElasticStress`
` [../]`

 ` [./eigen_strain]`
   ` type = ComputeVariableEigenstrain`
    `eigen_base = '-0.3464 0 0 -0.3092 0 -0.2116'`
   ` prefactor = h`
    `args = eta`
   ` eigenstrain_name = 'eigenstrain'`
  `[../]`

 ` [./elastic_free_energy_p]`
    `type = ElasticEnergyMaterial`
    `f_name = FE`
   ` args = 'eta'`
   ` derivative_order = 2`
 ` [../]`
  `#all free_Energy`
 ` [./free_energy]`
    `type = DerivativeSumMaterial`
  `  f_name = FA`
  `  sum_materials = 'FF FE'`
  `  args = 'eta'`
    `derivative_order = 2`
`  [../]`

How can I adjust my code?

**Comment by Krystalfang** [comment]
Hello, I have encountered the same problem, did you solve your problem?

**Comment by GiudGiud** [comment, marked as answer]
@dschwen @jiangwen84  is there some incompatible strain/stress materials in this list?

**Reply by biaogxb** [reply]
I guess it's because I didn't add displacement（disp_x \disp_y）,When I add the displacement term, the error report disappears, but there are some problems in the calculation result.
The results and input files without coupled mechanical stress are as follows,The result looks good.
![27_eta](https://user-images.githubusercontent.com/95063954/177449082-34876ae7-11ed-4dc3-a416-afb5e7ac684f.png)
[input1.txt](https://github.com/idaholab/moose/files/9050775/input1.txt)

When I coupled the mechanical stress, there were some problems in the result:(I don't know whether it is the problem of mechanical stress related parameters or the problem of mechanical stress coupling process)
![image](https://user-images.githubusercontent.com/95063954/177449462-1ac7dca4-f303-4f93-b2c3-4cb23ac8e8d6.png)
[input2.txt](https://github.com/idaholab/moose/files/9050794/input2.txt)

Could you give me some advice on coupling mechanical stress? I will be very grateful!

**Reply by GiudGiud** [reply]
I m not very knowledgable about this. 
This looks like unconverged results. Make sure nl_abs_tol and nl_rel_tol are both very low. They should both be several orders of magnitude below what seems acceptable in terms of convergence

**Reply by galibubc** [reply]
Hello @biaogxb,

Did you solve the problem? Can you please share how you solved it?

Musanna
